specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,205 @@
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@typespec/openapi";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using OpenAPI;
+
+@useAuth(AzureAuth)
+@service({
+  title: "Azure.SchemaRegistry",
+})
+
+// Supported operations.
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups?api-version={apiVersion}
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/$schemas/{id}?api-version={apiVersion}
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/?api-version={apiVersion}
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/{schemaVersion}?api-version={apiVersion}
+//  POST https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/:get-id?api-version={apiVersion}
+//  PUT https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}?api-version={apiVersion}
+@server(
+  "{endpoint}",
+  "The Schema Registry service endpoint.",
+  {
+    @doc("The Schema Registry service endpoint, for example 'my-namespace.servicebus.windows.net'.")
+    endpoint: url,
+  }
+)
+@versioned(ServiceApiVersions)
+namespace Azure.SchemaRegistry;
+
+@doc("Azure Core Version")
+enum ServiceApiVersions {
+  @useDependency(Azure.Core.Versions.v1_0_Preview_1)
+  @doc("Azure Schema Registry 2022-10 Version")
+  v2022_10: "2022-10",
```

Comment:
TODO: figure out how to add preview version.

Reply:
Let me know if you have any questions...the preview version can be added like this:
```suggestion
  v2022_10: "2022-10",
  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
  @doc("Azure Schema Registry 2023-09 Version")
  v2023_09: "2023-09",
```